docs/src/app/(with-menu)/hooks/usemapmanager/page.mdx

```diff
@@ -0,0 +1,93 @@
+export const metadata = {
+  title: 'useMapManager',
+  description: 'hook to get mapManager instance inside or outside RMap'
+};
+
+import { Flash } from "pentatrion-design/components/flash"
+
+# useMapManager
+
+Hook to get mapManager instance inside or outside RMap.
+
+This hook is used a lot internally to control the map instance. You probably 
+won't need to use it.
+
+## Usage inside RMap
+
+```tsx
+import { RMap, useMapManager, type MapManager } from "maplibre-react-components";
+
+function ChildComponent() {
+  // This component is inside RMap
+  // your mapManager instance is always available
+  const mapManager: MapManager = useMapManager();
+
+  return null;
+}
+
+export default function App() {
+  return (
+    <RMap>
+      <ChildComponent />
+    </RMap>
+  );
+}
+
+```
+
+<Flash color="yellow">
+If you use `useMapManager` inside `<RMap />` and you're using TypeScript it is better not to specify an `id` because 
+you will not have to handle the `null` case. But keep in mind that this is just for TypeScript because useMapManager is 
+**always** available when you are in `<RMap />`.
+</Flash>
+
+## Usage outside RMap
+
+If you want to access your map instance outside of RMap you must wrap your application in 
+a `<RMapContextProvider />` component. Moreover you must give an `id` to `<RMap />` and 
+provide this same `id` to `useMapManager`.
+
+<Flash color="orange">
+This might seem useless if your application contains only one map instance but this precaution allows TypeScript to return the correct type.
+</Flash>
+
+
+
+```tsx
+import { RMapContextProvider, RMap, useMapManager, type MapManager } from "maplibre-react-components";
+
+function ChildComponent() {
+  // This component contain <RMap /> or <RMap /> is a child of this component
+  // your MapLibre map instance is not always available
+  // you need to provide your <RMap /> id
```

Comment:
```suggestion
  // If you use `useMapManager` outside <RMap /> you need to provide the
  // `<RMap>`-id, since the MapLibre map instance is not always available.
```